dimod/include/dimod/quadratic_model.h

```diff
@@ -527,6 +530,112 @@ class BinaryQuadraticModel : public QuadraticModelBase<Bias, Index> {
         add_quadratic(dense, num_variables);
     }
 
+    /**
+     * Add the variables, interactions and biases from another BQM.
+     *
+     * The size of the updated BQM will be adjusted appropriately.
+     *
+     * If the other BQM does not have the same vartype, the biases are adjusted
+     * accordingly.
+     */
+    template <class B, class I>
+    void add_bqm(const BinaryQuadraticModel<B, I>& bqm) {
+        if (bqm.vartype() != vartype()) {
+            // we could do this without the copy, but for now let's just do
+            // it simply
+            auto bqm_copy = BinaryQuadraticModel<B, I>(bqm);
+            bqm_copy.change_vartype(vartype());
+            add_bqm(bqm_copy);
+            return;
+        }
+
+        // offset
+        base_type::offset() += bqm.offset();
+
+        // linear
+        if (bqm.num_variables() > base_type::num_variables()) {
+            resize(bqm.num_variables());
+        }
+        for (index_type v = 0; v < bqm.num_variables(); ++v) {
+            base_type::linear(v) += bqm.linear(v);
+        }
+
+        // quadratic
+        for (index_type v = 0; v < bqm.num_variables(); ++v) {
+            if (bqm.adj_[v].size() == 0) {
+                continue;
+            }
+
+            base_type::adj_[v].reserve(base_type::adj_[v].size() +
+                                       bqm.adj_[v].size());
+            for (auto it = bqm.adj_[v].cbegin(); it != bqm.adj_[v].cend();
```

Comment:
Oh I thought this was a vector, nevermind then. Since you would not have that lowlevel control over the elements of the neighborhood , I mean doing memcpy, etc.